README.md

```diff
@@ -6,6 +6,23 @@ REST APIs using Jersey, Jackson, Jetty, and Hibernate Validator.
 
 See the [`examples/`](examples/) directory for a simple demo service.
 
+Security
+--------
+The REST `Application` can support both http and https. https is disabled by default. Three modes are supported:
+http, https, and http+https. In the last mode, http and https are supported on different ports.
```

Comment:
Still scanning through the rest of the patch, but an initial thought: Kafka switched to a `listeners` strategy for specifying what protocol and port it would listen on. Would this make sense here as well? I would be worried about the potential for combinatoric explosion in the set of config values we'd need to check here if we add more variants (e.g. HTTP Basic Auth). Admittedly it's probably not that bad as we can always parse these (in fact, we could convert this setting to a `LIST` type instead and just separate with `,`). But then once you start adding in customization of the port being listened on, etc., it seems like it could get messier than just using the `listeners` config approach.